internal/app/machined/pkg/controllers/network/cmdline.go

```diff
@@ -163,6 +163,25 @@ func ParseCmdlineNetwork(cmdline *procfs.Cmdline) (CmdlineNetworking, error) {
 			}
 		}
 
+		// resolve enx* (with MAC address) to the actual interface name
+		if strings.HasPrefix(settings.LinkName, "enx") {
+			ifaces, _ := net.Interfaces() //nolint:errcheck // ignoring error here as ifaces will be empty
+			mac := strings.ToLower(strings.TrimPrefix(settings.LinkName, "enx"))
+
+			for _, iface := range ifaces {
+				ifaceMAC := strings.ReplaceAll(iface.HardwareAddr.String(), ":", "")
+				if ifaceMAC == mac {
+					settings.LinkName = iface.Name
+
+					break
+				}
+			}
+
+			if strings.HasPrefix(settings.LinkName, "enx") {
```

Comment:
need the if check again? since it's a;ready checking the same here https://github.com/siderolabs/talos/pull/6352/files#diff-9834688c7ece82096be9a1e985a0bff6a1889753c2a27e7829a159728f282bc3R167, 

seems this would always fail :thinking: , is `settings.LinkName` coming from machine config?

Reply:
The interface with MAC cannot be found, so I've checked it again...

Reply:
This can be rewritten, but the idea is that if we haven't done the substitution, there was no such MAC, so fail

Reply:
my bad, my brain wasn't grokking the code right yesterday :sweat_smile: